### PR TITLE
Merge ERC721_metadata into ERC721_base

### LINKED ISSUE #153: Define tokenURI methods in ERC721_base
ERC721_base should include tokenURI methods (like in OZ's Solidity implementation). Further, this implies that the `ERC721_token_uri` storage variable should be moved to ERC721_base. Finally, it might make sense to remove ERC721_Metadata altogether, as its only remaining function is to set the metadata magic value in the constructor.

### PR DESCRIPTION
**Pull request by andrew-fleming**
## Summary

This PR merges the ERC721_Metadata methods `tokenURI` and `setTokenURI` into ERC721_base. This PR proposes to initialize the the Metadata interface id in the ERC721_base constructor thus allowing for the removal of ERC721_Metadata altogether. Following the merge, all contracts (formerly 'frontend contracts') reflect the import changes. 

This PR also refactors the remaining snake case parameters to camel case for consistency. Further, this PR updates the ERC721 docs with the removal of the ERC721_Metadata_base contract. 

This resolves #153.

### CODE REVIEW

File: contracts/token/ERC721_Enumerable.cairo [hunk 1]
```diff
@@ -231,7 +244,18 @@ func burn{
         syscall_ptr: felt*, 
         range_check_ptr
     }(tokenId: Uint256):
-    Ownable_only_owner()
+    ERC721_only_token_owner(tokenId)
```
[the remark this answers is not in the document]
Comment (martriay) [reply contracts/token/ERC721_Enumerable.cairo:247]: Maybe instad of adding _Mintable_ to every preset, we could just clarify that every preset is Mintable on the docs

File: docs/ERC721.md [hunk 2]
```diff
@@ -248,7 +248,7 @@ The ERC721_Pausable preset creates a contract with pausable token transfers, min
 
 The ERC721_Metadata extension allows your smart contract to be interrogated for its name and for details about the assets which your NFTs represent.
 
-We follow OpenZeppelin's Solidity approach of integrating the Metadata methods `name`, `symbol`, and `tokenURI` into all ERC721 implementations. If preferred, a frontend contract can be created that imports from the ERC721_base directly (and does NOT include the Metadata extension). 
+We follow OpenZeppelin's Solidity approach of integrating the Metadata methods `name`, `symbol`, and `tokenURI` into all ERC721 implementations. If preferred, a contract can be created that does not import the Metadata methods from the ERC721_base library. Note that the IERC721_Metadata interface id should be removed from the constructor as well.
```
Comment (martriay) [inline docs/ERC721.md:251]: These should be code-formatted for consistency. I believe there's a couple extra missing ones.
```suggestion
We follow OpenZeppelin's Solidity approach of integrating the Metadata methods `name`, `symbol`, and `tokenURI` into all ERC721 implementations. If preferred, a contract can be created that does not import the Metadata methods from the `ERC721_base` library. Note that the `IERC721_Metadata` interface id should be removed from the constructor as well. 
```

Comment (martriay) [reply docs/ERC721.md:251]: Probably also mention that all presets are _Mintable_. They are, right?

**Review by martriay — APPROVED** [review]
[NOT APPROVED] I clicked the wrong button lol. Looking good, but also left some comments.

File: contracts/token/ERC721.cairo [hunk 3]
```diff
@@ -183,8 +192,9 @@ func setTokenURI{
         pedersen_ptr: HashBuiltin*, 
         syscall_ptr: felt*, 
         range_check_ptr
-    }(token_id: Uint256, token_uri: felt):
-    ERC721_Metadata_setTokenURI(token_id, token_uri)
+    }(tokenId: Uint256, tokenURI: felt):
+    Ownable_only_owner()
+    ERC721_setTokenURI(tokenId, tokenURI)
```
[the remark this answers is not in the document]
Comment (andrew-fleming) [reply contracts/token/ERC721.cairo:197]: Yeah, removing the pseudo preset sounds good. Let me know if we're in agreement here: our presets should be as follows:
- `ERC721_Burnable`
- `ERC721_Enumerable_Burnable`
- `ERC721_Pausable` (not burnable)

[the remark this answers is not in the document]
Comment (martriay) [reply contracts/token/ERC721.cairo:197]: Maybe let's call them _Mintable_. This should be extra clear.

* `ERC721_Mintable_Burnable`
* `ERC721_Enumerable_Mintable_Burnable`
* `ERC721_Mintable_Pausable`

File: docs/ERC721.md [hunk 4]
```diff
@@ -236,63 +241,63 @@ end
...
+### ERC721_Mintable_Burnable
 
-The ERC721_Mintable preset offers a quick and easy setup for creating NFTs. The contract owner can create and destroy tokens with the `mint` and `burn` methods. 
+The `ERC721_Mintable_Burnable` preset offers a quick and easy setup for creating NFTs. The contract owner can create tokens with `mint`, whereas token owners can destroy their tokens with `burn`. 
 
-### ERC721_Pausable
+### ERC721_Mintable_Pausable
 
-The ERC721_Pausable preset creates a contract with pausable token transfers, minting, and burning. This preset proves useful for scenarios such as preventing trades until the end of an evaluation period and having an emergency switch for freezing all token transfers in the event of a large bug. In this preset, only the owner can pause/unpause the contract.
+The `ERC721_Mintable_Pausable` preset creates a contract that enables the contract owner to `mint` new tokens as well as `pause` and `unpause` token transfers. This preset proves useful for scenarios such as preventing trades until the end of an evaluation period and having an emergency switch for freezing all token transfers in the event of a large bug.
```
Comment (martriay) [inline docs/ERC721.md:279]: I think this could be read as a bug in the preset

**Review by martriay — APPROVED** [review]
Good to go! Left a small comment

File: docs/ERC721.md (hunk 4 above)
Comment (andrew-fleming) [reply docs/ERC721.md:279]: Ahh I see what you mean. Good call

**Pull request merged**